pkg/ali/util/validation.ts

```diff
@@ -0,0 +1,63 @@
+import ipaddr from "ipaddr.js";
+
+function ipToLong(ip: string): number {
```

Comment:
Maybe I'm an ignoramus but I had NO idea what >>> did - could you add some comments explaining the IP chicanery in this file? Its a bit opaque